pravega/pravega
# Issue 3113: Kubernetes based system framework to deploy and test Zookeeper

**Change log description**  
Ensure the Zookeper system tests can run on a K8 Cluster. 

**Purpose of the change**  
Fixes the following aspects of #3113 
- enable zk deployment using zookeeper operator with ensure related system tests are also run on the K8 cluster.
- create a gradle task `startK8SystemTests` to run tests on a K8 Cluster.
- copies the test logs after the execution of test to `build/test-results` folder.

**What the code does**  
The code ensures zookeeper related system tests can be run on a K8 cluster. 
A gradle task `startK8SystemTests` has been created to run it.

```
gradle --info startK8SystemTests -PpravegaVersion=<pravegaVersion> 
-PrepoUrl=<repo where the build artifacts are present> --tests io.pravega.test.system.ZookeeperTest
```
( Note:  `uploadArchives` gradle task is used to upload the build artifacts. )

**How to verify it**  
Zookeeper tests should run in a K8 Environment.
Update: This has been tested on minikube and PKS deployments.
